include/sdf/ParserConfig.hh

```diff
@@ -188,13 +192,21 @@ class SDFORMAT_VISIBLE ParserConfig
   /// If the FindFileCallback provides a non-empty string, the URI will be
   /// stored in the DOM object, and the original (unresolved) URI will be
   /// stored in the underlying Element.
-  public: void SetStoreResovledURIs(bool _resolveURI);
+  public: void SetStoreResolvedURIs(bool _resolveURI);
 
   /// \brief Get the storeResolvedURIs flag value.
   /// \return True if the parser will attempt to resolve any URIs found and
   /// store them.  False to preserve original URIs
   public: bool StoreResolvedURIs() const;
 
+  /// \brief Set the path to search for URIs with relative path.
+  /// \param[in] _path Path to search.
+  public: void SetRelativeURISearchPath(const std::string &_path);
```

Comment:
I'm concerned that this will be confusing for users, especially since there's a chance for the value to be overwritten. It seems to me we're adding a public API to implement something that should be hidden from the user.

I propose an alternative that does not require adding anything new to ParserConfig:
Let's extend `resolveURI` to take an optional parent directory as an argument. Then it could do the following:
* If the URI is relative:
  1. First do the normal `sdf::findFile` call
  2. If that fails, prepend the parent directory and try again. (Or we can swap these steps)
 * If the URI is absolute, just do the normal `sdf::findFile`.

Reply:
thanks for suggestions. I implemented the proposed changes in [70e897e](https://github.com/gazebosim/sdformat/pull/1373/commits/70e897e875362e4e6924674d5699ed81fcdae55a). The `resolveURI` function now takes an optional set of search paths for resolving URIs.